File: arrays/rotateArray.js
const rotateArray = function(array, k) {
    const leng = array.length
    k = k%leng
    const temp = array.slice(leng-k)
    for(let i = leng-k-1;i>=0;i--){
        array[i+k] = array[i]
    }
    for (let i = 0; i< k; i++){
        array[i] = temp[i]
    }
    return array
}

const a = [1, 2, 3, 1, 2, 88, -1]
k=7
console.log(rotateArray(a, k))

File: arrays/rotate_array.py
def rotate_array(array, k):
    if len(array) == 0: return []
    k = k%len(array)
    temp = array[-k:]
    for i in reversed(range(0,len(array)-k)):
        array[i+k] = array[i]
    for i in range(len(temp)):
        array[i] = temp[i]
    return array

a = [3, 4, 5, 6, 7, 8]
print(rotate_array(a, 2))